lib/dataTransform.js

```diff
@@ -0,0 +1,114 @@
+const fs = require('fs'),
+  request = require('request'),
+  url = require('url'),
+  mime = require('mime'),
+  path = require('path'),
+  shell = require('shelljs');
+
+const download = (uri, filename) => {
+  return new Promise((resolve, reject) => {
+    request.head(uri, (err, res, body) => {
+      if (err) {
+        reject(err);
+      } else {
+        request(uri)
+          .pipe(fs.createWriteStream(filename))
+          .on('close', () => resolve(filename));
+      }
+    });
+  });
+};
+
+const filenameForUrl = uri => {
+  const filename = path.basename(url.parse(uri).pathname);
+  return `tmp/${filename}`;
+};
+
+const askForSlot = fileName => {
+  const params = { fileName: fileName, contentLength: fs.statSync(fileName)['size'], contentType: mime.getType(fileName) };
+  return new Promise((resolve, reject) => {
+    console.log(process.env.MARKETPLACE_TOKEN);
+    console.log(process.env.MARKETPLACE_URL);
+    console.log(DEPLOY_SERVICE_URL);
+    request.get(
```

Comment:
can you use request-promise?

returning all the "new Promise" and resolving is handled by default, handling errors is much easier, usage example in proxy.js